Consider ExportMixin.export_trigger_param in export_url

soerenbe opened the pull request:
The templatetag `export_url` currently dont consider the ExportMixin.export_trigger_param. I changed the implemenentation that it defaults to the parameter of the Table and not the hard coded  '_export'.

soerenbe commented:
I will have a look at it.

jieter commented:
You are right about the bug, I think I forgot to account for the custom case [when I created the tag](https://github.com/jieter/django-tables2/commit/8432a92632210eabd2fcd4df46c4835bea6a0960).

We could discuss if the template should be responsible to pass the value to the `{% export_url %}` tag, or that it's a valid assumption that a `view` key is present in the template context and the can take the value from the context, however, we should certainly consider the function base view use case. I don't like to force users to use the class based views, or to make `view` mandatory in the contexts, so I would still expect the tag to work without a `view` key in the context (or with view being something else0.

Shouldn't we do:

```python
if export_trigger_param is None and 'view' in context:
    export_trigger_param = getattr(context['view'], 'export_trigger_param', None)

export_trigger_param = export_trigger_param or '_export'
```

I think this makes it work with `ExportMixin`, with  function based view and with a view without an `export_trigger_param` attribute.

To fix the test, I'd say just add a 'view' key to the context, with an instance of an object based on `TemplateMixin` with a custom value for `trigger_export_param`:

```python
        # it should use the view.export_trigger_param attribute from the view, if available.
        class View(ExportMixin):
            export_trigger_param = "_do_export"

        template = Template("{% load django_tables2 %}{% export_url format %}")
        html = template.render(Context({"request": build_request("?q=foo"), "format": "xls", "view": View()}))
        self.assertEqual(dict(parse_qs(html[1:])), dict(parse_qs("q=foo&amp;_do_export=xls")))
```

soerenbe commented:
Hi,
I totally agree that my solution did not work with function bases views. Its a long time I used them. :-)

I updated the PR and incorporated you suggestions. I also added another unittest to cover the function based views.

If you have any feedback let me know.
